internal/worker/authenticationworker/utils.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package authenticationworker
+
+import (
+	"strings"
+
+	"github.com/juju/utils/v3/ssh"
+)
+
+const jujuEphemeralCommentPrefix = ssh.JujuCommentPrefix + "Ephemeral:"
+
+// ensureJujuEphemeralComment ensures that the key has a Juju ephemeral comment.
+// This is used to flush the ephemeral keys when the worker is restarted.
+func ensureJujuEphemeralComment(key string) string {
+	ak, err := ssh.ParseAuthorisedKey(key)
+	// Just return an invalid key as is.
```

Comment:
Can you justify this a bit more please. I agree we need to not break on invalid keys, but tell us more why this is possible/why we don't want to just error on a key.